tests/host_tools/network.py

```diff
@@ -129,14 +129,20 @@ def _init_connection(self):
             "ControlPersist=yes",
             *self.options,
             self.user_host,
-            "/usr/bin/true",
+            "true",
         ]
 
-        # don't set a low timeout here, because otherwise we might get into a race condition
-        # where ssh already forked off the persisted connection daemon, but gets killed here
-        # before exiting itself. In that case, self._control_path will exist, and the retry
-        # will hit the assert at the start of this function.
-        self._exec(establish_cmd, check=True)
+        try:
+            # don't set a low timeout here, because otherwise we might get into a race condition
+            # where ssh already forked off the persisted connection daemon, but gets killed here
+            # before exiting itself. In that case, self._control_path will exist, and the retry
+            # will hit the assert at the start of this function.
+            self._exec(establish_cmd, check=True)
+        except Exception:
+            # cleanup the control socket to allow for retries
+            if self._control_path.exists():
+                self._control_path.unlink()
+            raise
```

Comment:
Mh, if we do this then we can probably remove the assert altogether (and the comment, too). Although I think I slightly prefer not explicitly removing the path here. The idea behind this was that if the socket file ends up getting created, then we definitely have an SSH daemon process, and just retrying in this case will simply create more daemon processes. The assert prevent this from happening. Another option here would be to catch the exception and instead of unlinking, we do `self.close()`.

Reply:
fair point. I tested with just `self.close()` and it works as well. `close()` may fail, but in that case it would be retried anyways

Reply:
Yeah, but if `self._exec` fails and then `self.close()` as well, I'd honestly prefer us to error out completely, because something has fundamentally gone wrong then.

Mhhh... thinking about it now, maybe we should simply not retry in the case "exception and self._control_path.exists()". E.g. inside the `if` instead of unlink/close, we do `raise AbortRetriesAndFailException() from e` (name tbd), and in the retry decorator we make it not retry when it encounters this exception type (or potentially, only do that if _close_ also fails). That's pretty much the same behavior we have today, with the only difference being that instead of the assertion causing the abort (and thus hiding the actual cause), we get a better error message

Reply:
It starts to sound complicated for a fix that's a couple of lines. We cannot predict every possible failure mode, and know whether it's recoverable or not. If it ever fails like you describe, then we'd see it on the final retry and it would have the same error message. It would just take 20s more.

Reply:
that's fair enough